python/test/t_OptimizationAlgorithm_std.py

```diff
@@ -39,3 +39,34 @@
             else:
                 ref = 25.0 if ineq else 100.0
                 assert y[0] > ref
+
+
+def _exec(x):
+    if x[0] < 0.0 or x[0] > 1.0:
+        raise ValueError(f"Point {ot.Point(x)} not in bounds.")
+    c = 1.0 - x[0] ** 2
+    return [c]
+
+
+# check algorithm stays inside bounds
+costFunction = ot.PythonFunction(1, 1, _exec)
+problem = ot.OptimizationProblem(costFunction)
+bounds = ot.Interval([0.0], [1.0])
+problem.setBounds(bounds)
+for name in ot.OptimizationAlgorithm.GetAlgorithmNames():
+    algo = ot.OptimizationAlgorithm.Build(name)
+    try:
+        algo.setProblem(problem)
+        startingPoint = [0.5]
+        algo.setStartingPoint(startingPoint)
+    except Exception:
+        # not supported
+        continue
+    algo.setMaximumCallsNumber(100)
+    algo.setMaximumIterationNumber(100)
+    try:
+        algo.run()
+        assert algo.getResult().getStatus() == ot.OptimizationResult.SUCCEEDED
+        print(f"{name}: OK")
+    except Exception:
+        print(f"{name}:", algo.getResult().getStatusMessage())
```

Comment:
I don't understand this test. If the assert fails, then the exception is caught anyway? So how could this test fail?

Reply:
bonmin still failed here, resolved